### PR TITLE
pci-pch: Add a new plugin to detect the Intel DCI state

### PR DESCRIPTION
**Pull request by hughsie**
I don't have any systems where DCI can be enabled, so if you have anything I'd appreciate the sanity check.

### CODE REVIEW

File: plugins/pci-pch/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+PCI PCH
+=======
+
+Introduction
+------------
+
+This plugin checks if the PCH has DCI enabled. The result will be stored in an
```
Comment (superm1) [inline plugins/pci-pch/README.md:7]: Probably need to better describe what DCI is here.

File: src/fu-security-attr.c [hunk 2]
```diff
@@ -97,6 +97,10 @@ fu_security_attr_get_name (FwupdSecurityAttr *attr)
 		/* TRANSLATORS: Title: if the fwupd plugins are all present and correct */
 		return _("fwupd plugins");
 	}
+	if (g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_DCI) == 0) {
+		/* TRANSLATORS: Title: is DCI debugging is enabled */
```
Comment (superm1) [inline src/fu-security-attr.c:101]: If I don't know what this is, I think translators probably need to know too :)

File: plugins/pci-pch/README.md (hunk 1 above)
Comment (hughsie) [reply plugins/pci-pch/README.md]: Fixed. FYI, it's the double-ended USB-3 debug cable thing.

File: contrib/fwupd.spec.in [hunk 3]
```diff
@@ -369,6 +369,7 @@ mkdir -p --mode=0700 $RPM_BUILD_ROOT%{_localstatedir}/lib/fwupd/gnupg
 %{_libdir}/fwupd-plugins-3/libfu_plugin_optionrom.so
 %{_libdir}/fwupd-plugins-3/libfu_plugin_pci_bcr.so
 %{_libdir}/fwupd-plugins-3/libfu_plugin_pci_mei.so
+%{_libdir}/fwupd-plugins-3/libfu_plugin_pci_pch.so
```
Comment (superm1) [inline contrib/fwupd.spec.in:372]: I think this should probably be named pci_dci no?  Unless you're thinking to get other attributes from this later too.

**Comment by superm1** [comment]
> I don't have any systems where DCI can be enabled, so if you have anything I'd appreciate the sanity check.

The pre-production systems on my side show this disabled, but you might need a debug BIOS in the first place with it enabled, I don't know.

File: contrib/fwupd.spec.in (hunk 3 above)
Comment (hughsie) [reply contrib/fwupd.spec.in:372]: I think the PCH is a very interesting device with a lot more to measure.

**Comment by hughsie** [comment]
> but you might need a debug BIOS in the first place with it enabled

Will ask on twitter.

**Comment by hughsie** [comment]
@superm1 can you test your Dell systems with the latest version of this branch? I'm guessing they should be also "locked" but some of my Lenovo systems were not so lucky...

**Comment by superm1** [comment]
> @superm1 can you test your Dell systems with the latest version of this branch? I'm guessing they should be also "locked" but some of my Lenovo systems were not so lucky...

I've only got some pre-production stuff handy so it might not be a valid test.  The pre-production stuff shows it as "unlocked".

File: src/fu-security-attr.c [hunk 4]
```diff
@@ -129,6 +129,12 @@ fu_security_attr_get_name (FwupdSecurityAttr *attr)
 		/* TRANSLATORS: Title: if the fwupd plugins are all present and correct */
 		return g_strdup (_("fwupd plugins"));
 	}
+	if (g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_DCI_ENABLED) == 0 ||
+	    g_strcmp0 (appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_DCI_LOCKED) == 0) {
+		/* TRANSLATORS: Title: Direct Connect Interface (DCI) allows
+		 * debugging of Intel processors using the USB3 port */
+		return g_strdup (_("DCI Debugger"));
```
Comment (superm1) [inline src/fu-security-attr.c:136]: This should probably be prefixed with Intel (like CET was)

**Comment by hughsie** [comment]
>  The pre-production stuff shows it as "unlocked".

That might be completely valid, or I might be reading the entirely wrong register and just getting all zeros.

**Comment by superm1** [comment]
> > The pre-production stuff shows it as "unlocked".
> 
> That might be completely valid, or I might be reading the entirely wrong register and just getting all zeros.

I'm not really sure here.  This might be a case that it needs Intel's help to confirm it.

**Comment by hughsie** [comment]
>  This might be a case that it needs Intel's help to confirm it.

The advice is to use the MSR interface. Will close this PR and switch to using that. At least with my initial testing my Lenovo hardware is coming up `Locked` so that's more encouraging....

**Comment by superm1** [comment]
> > This might be a case that it needs Intel's help to confirm it.
> 
> The advice is to use the MSR interface. Will close this PR and switch to using that. At least with my initial testing my Lenovo hardware is coming up `Locked` so that's more encouraging....

Doesn't the MSR interface not work in kernel lockdown?

**Comment by hughsie** [comment]
> Doesn't the MSR interface not work in kernel lockdown?

Seems to work here, although I'm only _reading_...

**Comment by superm1** [comment]
> > Doesn't the MSR interface not work in kernel lockdown?
> 
> Seems to work here, although I'm only _reading_...

That might be why, I do recall specifically alteration of MSR is what's supposed to be locked down.

**Pull request closed without merging**